src/libraries/System.Formats.Nrbf/src/System/Formats/Nrbf/ArraySinglePrimitiveRecord.cs

```diff
@@ -94,7 +86,7 @@ internal static IReadOnlyList<T> DecodePrimitiveTypes(BinaryReader reader, int c
 #if NET
         reader.BaseStream.ReadExactly(resultAsBytes);
 #else
-        byte[] bytes = ArrayPool<byte>.Shared.Rent(Math.Min(count * Unsafe.SizeOf<T>(), 256_000));
+        byte[] bytes = ArrayPool<byte>.Shared.Rent((int)Math.Min(requiredBytes, 256_000));
```

Comment:
this fixes an int32 overflow: `large number of elements * sizeof(long)`

Reply:
Why does `requiredBytes` not have a similar problem? Shouldn't that calculation be checked?

Reply:
When it's calculated:

https://github.com/dotnet/runtime/blob/a3365e4c442a728bc82e59d5583cf83bc6d31d72/src/libraries/System.Formats.Nrbf/src/System/Formats/Nrbf/ArraySinglePrimitiveRecord.cs#L64-L67

We know that count is less than `Array.MaxLength` and `T` is limited to a list of natively supported primitive types. So the max value we can get here is `Array.MaxLength * 8` which won't overflow `Int64`. Then `Math.Min(long,  256_000)` is going to return `256_000` which can be caster to `Int32` without an overflow.